src/mono/mono/eventpipe/ds-rt-mono.h

```diff
@@ -244,6 +244,56 @@ ds_rt_profiler_startup (DiagnosticsStartupProfilerCommandPayload *payload)
 	return DS_IPC_E_NOTSUPPORTED;
 }
 
+/*
+* Environment variables
+*/
+
+static
+uint32_t
+ds_rt_set_environment_variable (const ep_char16_t *name, const ep_char16_t *value)
```

Comment:
Could we make this take a delimited list of names and values? That way we could set multiple in one command?

Reply:
It could, but I think it would introduce some complication that isn't really necessary. We'd have to use a delimiter and then either wouldn't support variables with the delimiter or have to introduce escaping logic which introduces complexity. It's definitely doable, but I don't think people are going to be setting hundreds of environment variables so I expect the perf win isn't all that critical

Reply:
"Reverse" connections (runtime connects to diagnostic tool's listener) _might_ be slower. I'd be interested in the perf of that if you can set that up.

Reply:
> "Reverse" connections (runtime connects to diagnostic tool's listener) might be slower

Does that really matter as long as the runtime is blocked from making further progress in startup under reverse mode?

Reply:
If it is significantly slower, then it very much matters. There's a lot more blocking and waiting involved in the "reverse" connection. And given these are being added to support setting up a profiler during suspend (which is more likely than not to happen when "reverse" connected), I think the data should be known for the scenario that is trying to be supported. The current numbers are good for "forward" connections, which if testing in the profiler scenario, is not going to be a common configuration.

Reply:
I am assuming the perf numbers are for "forward" but @davmason can clarify. If these numbers are for "reverse", then those look great.

Reply:
The numbers I posted were collected by setting up a reverse server, having the runtime attach, and then while sitting with the connection open issuing 1,000 set commands. The timing is just for the set commands and not for the whole setting up a reverse server. What I was saying is that it will increase the existing time by ~100ms if you want to set 1,000 environment variables